cetlvast/cmake/modules/Findgoogletest.cmake

```diff
@@ -0,0 +1,408 @@
+#
+# Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+# Copyright Amazon.com Inc. or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+
+if(NOT TARGET gtest_main)
+
+enable_testing()
+
+include(FetchContent)
+include(FindPackageHandleStandardArgs)
+
+set(googletest_GIT_REPOSITORY "https://github.com/google/googletest.git")
+set(googletest_GIT_TAG "b796f7d44681514f58a683a3a71ff17c94edb0c1")
+
+FetchContent_Declare(
+    googletest
+    GIT_REPOSITORY  ${googletest_GIT_REPOSITORY}
+    GIT_TAG         ${googletest_GIT_TAG}
+)
+
+# The automatic management of the <lowercase>_POPULATED name appears to be broken in
+# cmake 3.21 and earlier.  This workaround may not be needed after 3.24.
+get_property(googletest_POPULATED GLOBAL PROPERTY googletest_POPULATED)
+
+if(NOT googletest_POPULATED)
+
+    if (NOT FETCHCONTENT_SOURCE_DIR_googletest)
+        set(FETCHCONTENT_SOURCE_DIR_googletest ${CETLVAST_EXTERNAL_ROOT}/googletest)
+    endif()
+
+    if (NOT ${FETCHCONTENT_FULLY_DISCONNECTED})
```

Comment:
I want to be able to point it at a local GoogleTest installation instead of downloading it from Git.  Looks like this is set up so it's possible, that right?

Reply:
I could wire in a redirect but you can fetch googletest once then check it all into a branch to avoid doing that elsewhere. In fact, I'm planning on releasing CETL with all of these external dependencies already fetched so working from CETL releases you wouldn't need to be online.